core/src/main/scala/org/apache/spark/rdd/CoalescedRDD.scala

```diff
@@ -121,6 +121,21 @@ private[spark] class CoalescedRDD[T: ClassTag](
 }
 
 /**
+ * A PartitionCoalescer defines how to coalesce the partitions of a given RDD.
+ */
+trait PartitionCoalescer {
+
+  /**
+   * Coalesce the partitions of the given RDD.
+   *
+   * @param maxPartitions the maximum number of partitions to have after coalescing
+   * @param parent the parent RDD whose partitions to coalesce
+   * @return a grouping of the parent RDD's partitions
+   */
+  def coalesce(maxPartitions: Int, parent: RDD[_]): Array[PartitionGroup]
```

Comment:
can you document a little bit more the semantics of the return? 

i think it's better to say explicitly each element in the returned array represents one rdd partition post-coalescing.